src/lib/Invitation/Persistence/DoctrineGateway.php

```diff
@@ -145,18 +147,86 @@ private function getSelectQuery(): QueryBuilder
             );
     }
 
-    public function markAsUsed(string $hash): void
+    public function findInvitations(?InvitationFilter $filter = null): array
     {
-        $query = $this->connection->createQueryBuilder();
-        $query
-            ->update(self::TABLE_USER_INVITATIONS)
-            ->set('used', 1)
-            ->where(
+        $query = $this->getSelectQuery();
+        $expr = $query->expr();
+
+        if ($filter === null) {
+            $statement = $query->execute();
+
+            return $statement->fetchAllAssociative();
+        }
+
+        $filters = [];
+        if ($filter->getRole() !== null) {
+            $filters[] =
                 $query->expr()->eq(
-                    'hash',
-                    $query->createPositionalParameter($hash)
-                )
+                    't2.role_id',
+                    $query->createPositionalParameter($filter->getRole()->id)
+                );
+        }
+
+        if ($filter->getUserGroup() !== null) {
+            $filters[] =
+                $query->expr()->eq(
+                    't2.user_group_id',
+                    $query->createPositionalParameter($filter->getUserGroup()->id)
+                );
+        }
+
+        if ($filter->getIsUsed() !== null) {
+            $filters[] =
+                $query->expr()->eq(
+                    't1.used',
+                    $query->createPositionalParameter($filter->getIsUsed())
+                );
+        }
+
+        if (!empty($filters)) {
+            $query->where(
```

Comment:
```suggestion
            $query->andWhere(
```
Additionally, you could have just added those conditions using `andWhere` instead of storing them in a variable :sweat_smile:

Reply:
Please, move that review to relevant PR.
https://github.com/ibexa/user/pull/41/files#

Reply:
Would be easier to spot that those changes come from a different PR **if** the PR base would be selected to the actual base (which automatically changes if the original branch is merged btw).